src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/GroovyCallSiteSelector.java

```diff
@@ -102,6 +85,33 @@ private static boolean matches(@Nonnull Class<?>[] parameterTypes, @Nonnull Obje
     }
 
     /**
+     * Translates a method parameter list with varargs possibly spliced into the end into the actual parameters to be passed to the JVM call.
+     */
+    private static Object[] parametersForVarargs(Class<?>[] parameterTypes, Object[] parameters) {
+        int fixedLen = parameterTypes.length - 1;
+        Class<?> componentType = parameterTypes[fixedLen].getComponentType();
+        assert componentType != null;
+        int arrayLength = parameters.length - fixedLen;
+        if (arrayLength >= 0) {
+            if (arrayLength == 1 && parameterTypes[fixedLen].isInstance(parameters[fixedLen])) {
+                // not a varargs call
+                return parameters;
+            } else {
+                Object array = Array.newInstance(componentType, arrayLength);
+                for (int i = 0; i < arrayLength; i++) {
+                    Array.set(array, i, parameters[fixedLen + i]);
+                }
+                Object[] parameters2 = new Object[fixedLen + 1];
+                System.arraycopy(parameters, 0, parameters2, 0, fixedLen);
```

Comment:
Super nitpicky 🐜 - can create and copy parameters2 with a single copyOfRange call (null will be placed in fixedLen until set).